torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -244,6 +244,7 @@ void ProcessGroupAgent::start() {
 void ProcessGroupAgent::shutdown() {
   LOG(INFO) << "Shutting down ProcessGroupAgent on rank " << pg_->getRank()
             << ".";
+  RpcAgent::shutdown();
   std::unique_lock<std::mutex> lock{futureMutex_};
   if (!rpcRunning_.exchange(false)) {
```

Comment:
btw, reading the code, it's easy to confuse rpcRunning_ from rpcAgentRunning_.
Especially because rpcAgentRunning_ is protected and visible in this scope 

Not sure exactly what to say (maybe just making RpcAgent's var private might help?) but this sort of thing could lead to future bugs, since the var names are easy to confuse

Reply:
I actually plan on consolidating `rpcRunning_` and `rpcAgentRunning_` in a future PR. Here's the issue to track: #32963

Reply:
Sounds good. Just be careful how you do exchange(false) during shutdown, especially when superclass's shutdown is at the end.